src/main/scala/codacy/docker/api/Tool.scala

```diff
@@ -3,7 +3,8 @@ package codacy.docker.api
 import scala.util.Try
 
 trait Tool {
-  def apply(source: Source.Directory, configuration: Option[List[Pattern.Definition]], files: Option[Set[Source.File]])
```

Comment:
There are implicits in codacy/codacy-engine-seed-scala that might discard the options parameter if we do this.